diff --git a/core/resolver/agoresolver.cpp b/core/resolver/agoresolver.cpp
index 833ec0f..27fa12a 100644
--- a/core/resolver/agoresolver.cpp
+++ b/core/resolver/agoresolver.cpp
@@ -428,7 +428,7 @@ int main(int argc, char **argv) {
 	}
 
 //	clog << agocontrol::kLogDebug << "reading inventory" << std::endl;
-	inv = new Inventory(CONFDIR "/db/inventory.db");
+	inv = new Inventory(LOCALSTATEDIR "/db/inventory.db");
 
 	variables = jsonFileToVariantMap(VARIABLESMAPFILE);
 	if (persistence) loadDevicemap();
diff --git a/core/resolver/inventory.cpp b/core/resolver/inventory.cpp
index 64f8414..28cfde5 100644
--- a/core/resolver/inventory.cpp
+++ b/core/resolver/inventory.cpp
@@ -190,6 +190,14 @@ string Inventory::getfirst(const char *query, int n, ...) {
 		case SQLITE_ROW:
 			if (sqlite3_column_type(stmt, 0) == SQLITE_TEXT) result =string( (const char*)sqlite3_column_text(stmt, 0));
 			break;
+		case SQLITE_CANTOPEN:
+			fprintf(stderr, "Unable to open db. Please ensure the db file and its containing directory are writeable.\n");
+			break;
+		case SQLITE_DONE:
+			break;
+	        default:
+			printf("Unhandled return %d\n", rc);
+			break;
 	}
 
     va_end(args);
